operator-pipeline-images/operatorcert/integration/external_tools.py

```diff
@@ -0,0 +1,197 @@
+"""
+Utility classes to run external tools
+"""
+
+import base64
+import json
+import logging
+import subprocess
+import tempfile
+from os import PathLike
+from pathlib import Path
+from typing import Mapping, Optional, Sequence, TypeAlias
+
+LOGGER = logging.getLogger("operator-cert")
+
+
+CommandArg: TypeAlias = str | PathLike[str]
+
+
+class Secret(str):
```

Comment:
How does this class work? What mechanism is used to prevent printing the secret? Is this due to __repr__ and __str__ are different now?